### PR TITLE
Hostile Automata (Baseline)

### PR DESCRIPTION
**Pull request by klaasjared**
This adds the basic code needed for Hostile Automata - an update that aims to eliminate the Hivebots as well as Human NPCs such as Mercs. So far this is a major WIP, but the baseline for the update is working (kinda) and can be added and expanded upon to add new and exciting hostiles to POIs and Gateway maps.

So far this PR adds the following:
-Basic Automaton
basic
-Ranged Automaton
ranged
-Commie Bot
commie
-Commie Officer Bot
officer
-Malf Drone
drone
-Syndicate Sentry
sentry
-Syndicate Guard
guard
-Syndicate Tank
tank
-Two Custom Dead Sprites
dead1
dead2

Again, this is still a WIP and it is subject to change at any given time.

Or get shot down immediately - who knows.

**_EDIT:_**

A lot of people seem to be confused on what the state of this PR is and why it exists.

This PR is 'baseline code' or a foundation - something to build off of. The robots listed above do not and likely will not be placed in any map spawns or POIs and are strictly for testing and admin spawn.

As of right now the code is working and ready to be pushed into the master which is why the PR is here and not in a draft state.

I WILL be replacing the hivebots with new robot sprites. They will act and function almost if not just like the current hivebots. I'm making this new code separate from the existing code to prevent a "TF2 Coconut" situation.

I have plans to add custom robotic sprites in the near future - but work, school, and general life is making things difficult. I will be working on the new robots on and off as time goes on.

Thank you for understanding.

### CODE REVIEW

File: code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+	These surplus robot was widely used by stations, colonies, and bases for dumb-labor such as janitorial, \
+	logistical, and even simple area security. These older models would later be phased out for more favorable \
+	modernized technology as the year went by but abandoned establishments and the older stations may still \
+	use these models for basic labor.\
+	<br>\
+	These robots were made cheap and from cheap material and thus aren't too much of a threat in small numbers - \
+	however, travellers are cautioned against engaging larger groups as even the most simple of units in \
+	large numbers can overwhelm even the most prepared people. \
+	<br>\
+	Depending on what sort of work or tasks the automaton may be assigned these botsm their Intelligence level can \
```
Comment (TheFurryFeline) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm:18]:
```suggestion
	Depending on what sort of work or tasks the automaton may be assigned these bots, their Intelligence level can \
```

File: code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	value = CATALOGUER_REWARD_TRIVIAL
+
+/datum/category_item/catalogue/technology/drone/automaton/syndi
+	name = "Drone - Syndicate Securibot"
+	desc = "Classification: R04 V9 Securibot\
+	<br><br>\
+	This is an old, unused line of security automatons whose designs were stolen from NanoTrasen in the early 2280s. \
+	The design was meant to replace living security with a cheap and effective unit that could serve to protect \
+	establisments and stations around known space. The Syndicate stole the designs and adapted the Gamma Level framework \
+	to meet their need for automated security in thier own bases as well as frontline soldiers that could be \
```
Comment (TheFurryFeline) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm:31]:
```suggestion
	establishments and stations around known space. The Syndicate stole the designs and adapted the Gamma Level framework \
	to meet their need for automated security in their own bases as well as frontline soldiers that could be \
```

File: code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+/mob/living/simple_mob/mechanical/automaton/commie
+	name = "Communist Robot"
+	desc = "A mass produced robotic unit dressed in communist garb."
+
+	icon_state = "commie_bot"
+	icon_living = "commie_bot"
+
+/mob/living/simple_mob/mechanical/automaton/comofficer
+	name = "Communist Officer Robot"
+	desc = "A mass produced robotic unit dressed in communist garb. This one appears to be more armored and has an advanced intergrated laser."
```
Comment (TheFurryFeline) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm:124]:
```suggestion
	desc = "A mass produced robotic unit dressed in communist garb. This one appears to be more armored and has an advanced integrated laser."
```

File: code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	maxHealth = 100
+	health = 100
+	movement_cooldown = 5
+	melee_damage_lower = 10
+	melee_damage_upper = 10
+	attacktext = list("bashed", "whacked", "hit", "beaten")
+
+/mob/living/simple_mob/mechanical/automaton/syndi/ranged
+	name = "Syndicate Securibot Guard"
+	desc = "A mass produced robotic unit that's been programmed to protect a given area. It has an intergrated photonic weapon system."
```
Comment (TheFurryFeline) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm:193]:
```suggestion
	desc = "A mass produced robotic unit that's been programmed to protect a given area. It has an integrated photonic weapon system."
```

File: code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	faction = "syndicate"
+	ai_holder_type = /datum/ai_holder/simple_mob/mechanical/automaton/syndi
+
+	maxHealth = 100
+	health = 100
+	movement_cooldown = 8
+
+/mob/living/simple_mob/mechanical/automaton/syndi/tank
+	name = "Syndicate Securibot Tank"
+	desc = "A cyborg shell with no higher level AI installed, instead it's been programmed to protect a given area. It has a heavy photonic weapon system built in to the body."
```
Comment (TheFurryFeline) [inline code/modules/mob/living/simple_mob/subtypes/mechanical/hivebot/hostile_automata.dm:213]:
```suggestion
	desc = "A cyborg shell with no higher level AI installed, instead it's been programmed to protect a given area. It has a heavy photonic weapon system built into the body."
```

**Review by TheFurryFeline — COMMENTED** [review]
Minor spelling fixes

**Comment by dwinters99** [comment]
This is good, but I'll be sad to see hivebots go. They are just- too classic.

**Comment by KillianKirilenko** [comment]
Not sold on these at all - they're OG cyborgs (some of which have clothes or weapons), and the basic ranged variant has maybe a *single* pixel of difference from the basic melee variant? Explicitly labelling things as having any connection to anything "Syndicate" seems like a bad idea as well.

I'm also not sure why hivebots need to be replaced, exactly. I feel like it would be better to take the hivebots and flesh them out with more distinctive designs and more unique sprites, rather than replace them with "new" mobs using generic, old sprites that have had a few things added onto thtem.

**Comment by lm40** [comment]
Yeah I kinda agree with most of the above points. Like, no need to get rid of the hivebots if the replacement is basically repurposed cyborg sprites, I'm not attached to their appearance otherwise but at least they're distinctive.
I also agree that labeling things 'syndicate' is iffy. At least do something like 'ominous sentry' or 'suspicious sentry' or whatever descriptors we use for things that are syndicate/traitor gear.

**Pull request closed without merging**